src/focus.service.ts

```diff
@@ -362,46 +332,35 @@ export class FocusService {
       return;
     }
 
-    this.rescroll(next, scrollSpeed, this.root);
-
     const attached = selected && isNodeAttached(selected, this.root);
 
-    this.parents = [];
-
     if (attached) {
       // Find the common ancestor of the next and currently selected element.
       // Remove selected classes in the current subtree, and add selected
       // classes in the other subtree. Trigger focus changes on every
       // element that we touch.
-      selected.blur();
-
       const common = getCommonAncestor(next, selected);
-      selected.classList.remove(cssClass.direct);
```

Comment:
I see what you're getting at, but don't we run into the same problem there? Determining the source of the focus is not trivial, and in some cases it may be impossible with certain accessibility software. I'd rather a unified style that we don't need to change whether we're using a keyboard, mouse, or any other device to navigate.

Regardless, determining that it outside of the scope of arcade-machine, as keyboard navigation is possible without using arcade-machine at all. A system that determines the source of a focus event would likely be a standalone module.